Patch by schwardt@univention.de (Wed Sep 11 16:12:13 2013 +0200):

use sbsigntool instead of pesign



--- shim-0.2.ORIG/Makefile
+++ shim-0.2/Makefile
@@ -68,10 +68,10 @@ version.c : version.c.in
 		< version.c.in > version.c
 
 certdb/secmod.db: shim.crt
-	-mkdir certdb
-	certutil -A -n 'my CA' -d certdb/ -t CT,CT,CT -i ca.crt
-	pk12util -d certdb/ -i shim.p12 -W "" -K ""
-	certutil -d certdb/ -A -i shim.crt -n shim -t u
+	#-mkdir certdb
+	#certutil -A -n 'my CA' -d certdb/ -t CT,CT,CT -i ca.crt
+	#pk12util -d certdb/ -i shim.p12 -W "" -K ""
+	#certutil -d certdb/ -A -i shim.crt -n shim -t u
 
 shim.o: $(SOURCES) shim_cert.h
 
@@ -114,7 +114,8 @@ lib/lib.a:
 		--target=efi-app-$(ARCH) $^ $@.debug
 
 %.efi.signed: %.efi certdb/secmod.db
-	pesign -n certdb -i $< -c "shim" -s -o $@ -f
+	#pesign -n certdb -i $< -c "shim" -s -o $@ -f
+	sbsign --key shim.key --cert shim.crt --output $@ $<
 
 clean:
 	$(MAKE) -C Cryptlib clean
